### PR TITLE
✨ Implementing real recent guides and tutorials section

### PR DESCRIPTION
**Pull request by sebil**
**With this PR the teasers of the "Recent Guides & Tutorials" section getting really ordered by date and time.**

There is now a new gulp task for the pipeline that starts on `build` and `bootstrap` which is looking for all pages inside `/content/amp-dev/documentation/guides-and-tutorials/` and writes them together with the `committer date` in `ISO 8601-like format` inside a `.yaml` file.

The list is ordered in descending order and looks like this:

```yaml
- path: /content/amp-dev/documentation/guides-and-tutorials/index.html
  date: '2019-11-26 17:12:32 +0100'
- path: /content/amp-dev/documentation/guides-and-tutorials/contribute/translations.md
  date: '2019-11-18 11:31:55 -0800'
- path: /content/amp-dev/documentation/guides-and-tutorials/learn/amp-actions-and-events.md
  date: '2019-11-18 02:07:17 -0800'
- path: /content/amp-dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cache-urls.md
  date: '2019-11-14 15:53:59 -0800'
- path: /content/amp-dev/documentation/guides-and-tutorials/start/visual_story/congratulations.md
  date: '2019-11-12 07:21:48 -0800'
- path: /content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/ad-integration-guide.md
  date: '2019-11-08 10:50:46 -0800'
...
```

Here is a screenshot of the section:

Screenshot 2019-11-26 at 17 26 43

### CODE REVIEW

File: platform/lib/pipeline/recentGuides.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      `${PATH_ABSOLUTE}/**/*.{md,html}`,
+      `!${PATH_ABSOLUTE}/**/*@*.{md,html}`,
+    ]);
+
+    log.start('Refreshing recent guides list ...');
+
+    stream = stream.pipe(through.obj(async (file, encoding, callback) => {
+      if (!file.isDirectory() && !file.isNull()) {
+        // Create file path and get file date with git log function
+        const filePath = PATH_RELATIVE + file.relative;
```
Comment (sebastianbenz) [inline platform/lib/pipeline/recentGuides.js]: use `path.join`

File: platform/lib/pipeline/recentGuides.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+const yaml = require('js-yaml');
+const through = require('through2');
+const git = require('@lib/utils/git');
+const project = require('@lib/utils/project');
+const {Signale} = require('signale');
+const log = new Signale({'scope': 'Recent Guides'});
+
+// Paths of guides pages; Relative for yaml file; Absolute for git log function
+const PATH_RELATIVE = '/content/amp-dev/documentation/guides-and-tutorials/';
+const PATH_ABSOLUTE = project.paths.GROW_POD + PATH_RELATIVE;
```
Comment (sebastianbenz) [inline platform/lib/pipeline/recentGuides.js]: please use path.join

File: platform/lib/pipeline/recentGuides.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+const fs = require('fs');
+const yaml = require('js-yaml');
+const through = require('through2');
+const git = require('@lib/utils/git');
+const project = require('@lib/utils/project');
+const {Signale} = require('signale');
+const log = new Signale({'scope': 'Recent Guides'});
+
+// Paths of guides pages; Relative for yaml file; Absolute for git log function
+const PATH_RELATIVE = '/content/amp-dev/documentation/guides-and-tutorials/';
```
Comment (sebastianbenz) [inline platform/lib/pipeline/recentGuides.js]: Please move this to `utils/project.js`

File: platform/lib/pipeline/recentGuides.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+const PATH_ABSOLUTE = project.paths.GROW_POD + PATH_RELATIVE;
+
+// Where to save the list to
+const DEST_FILE = 'pages/shared/data/recent-guides.yaml';
```
Comment (sebastianbenz) [inline platform/lib/pipeline/recentGuides.js]: Please move this to `utils/project.js`

File: platform/lib/utils/git.js [hunk 5]
```diff
@@ -19,3 +19,6 @@ const {execSync} = require('child_process');
 module.exports.version = execSync('git log -1 --pretty=format:%h ').toString().trim();
 module.exports.message = execSync('git log -1 --pretty=%B --no-merges').toString().trim();
 module.exports.user = execSync('git config user.name').toString().trim();
+module.exports.committerDate = (path) => {
+  return execSync(`git log -1 --pretty=format:%ci ${path}`).toString().trim();
```
Comment (sebastianbenz) [inline platform/lib/utils/git.js]: This sorts by latest update (which could be a fixed typo). I think it'd be better to use the time the file was added to git.

**Comment by sebil** [comment]
Thanks for the feedback, @sebastianbenz, everything done.
For the date I used the very first author date instead of the last committer date now.

**Pull request merged**